app/knights/equipment.py

```diff
@@ -0,0 +1,33 @@
+class Equipment:
+    """Base class for all equipment types."""
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+
+class Weapon(Equipment):
+    """Represents a weapon with a power attribute."""
+
+    def __init__(self, name: str, power: int) -> None:
+        super().__init__(name)
+        self.power = power
+
+
+class Armour(Equipment):
+    """Represents a piece of armor with a protection value."""
+
+    def __init__(self, part: int, protection: str) -> None:
+        super().__init__(part)
```

Comment:
The `Armour` class constructor incorrectly uses the `part` parameter as the `name` parameter for the `Equipment` base class. This should be corrected to ensure proper initialization. Consider changing the `part` parameter to a more appropriate name or adjusting the constructor to handle the `name` parameter correctly.